### ISSUE TITLE
Os Error: too many files open + Error sending Request

### ISSUE THREAD

**Issue by emilvincentz** [acterglobal/a3#1788]
The app has started to crash + not loading spaces or images in chat properly + bug report also not working.

![Image](https://github.com/acterglobal/a3/assets/77483554/cee629b3-7fc8-4ea3-b65e-f35984e3a99d)

![Image](https://github.com/acterglobal/a3/assets/77483554/79665d0d-3ab5-4624-9b11-15602d7b5949)

![Image](https://github.com/acterglobal/a3/assets/77483554/8de1ea51-f4dc-4d88-99a6-01ddcb601940)

**Comment by emilvincentz** [comment]
Might also be connected to this loading errror:


![Image](https://github.com/acterglobal/a3/assets/77483554/7574e368-e4c8-4466-910f-c294ef602272)

**Comment by emilvincentz** [comment]
maybe related to this? 
error loading image in chat #1261

**Comment by gnunicorn** [comment]
looks like we are leaking some file descriptor(s) ... there is plenty of reasons why that could be and maybe we just want to "resolves" this problem for now by reacting on that particular error code with prompting the user to restart the app ... it is hard for us to investigate that at the moment ..

**Comment by emilvincentz** [comment]
This now happens all the time and doesn't help with restarting the app

**Comment by emilvincentz** [comment]
![image](https://github.com/acterglobal/a3/assets/77483554/0bb32465-4d60-4587-9618-8767c7b4d5ee)

![image](https://github.com/acterglobal/a3/assets/77483554/8960b3a8-aae9-4315-86cc-e3b5a934ba0e)

**Comment by emilvincentz** [comment]
well the last one Im not 100% sure of. All the loading error just started appearing at ones, along with the crashes and fail to bug report too, so thought they might be all connected, but maybe not

**Comment by emilvincentz** [comment]
It seems that they somehow are related. At least the two main messages that keeps appearing is the "Error sending request" and "Too many files open".
Below you can see that error messages changes between these two error messages, compared to the ones above.
Also when trying to bug report and error message shows, the app freezes.



![Image](https://github.com/acterglobal/a3/assets/77483554/e5e1f284-2a89-4f18-bfc5-d258416e953e)
![Image](https://github.com/acterglobal/a3/assets/77483554/931dc72e-7e3e-49ff-bf28-5d8835a47086)
![Image](https://github.com/acterglobal/a3/assets/77483554/f436b653-d0ea-4506-ab4f-b31b5289fec0)

**Comment by emilvincentz** [comment]
On IOS these errors are now happening all the time

**Comment by gnunicorn** [comment]
> On IOS these errors are now happening all the time

Is that on release or nightly? I'll investigate what is going on ...

**Comment by emilvincentz** [comment]
just tested, release on App Store works fine. Only on the latest couple of nightlies where this happens

**Comment by gnunicorn** [comment]
okay. thanks for the info. according to logs of https://github.com/acterglobal/a3-meta/issues/264 it already happens almost immediately after start/sync ... which is severely odd. I suspect the iOS spawn-process is doing something bad there that leads to a FO explosion...

**Comment by emilvincentz** [comment]
This still happens on latest nightly

**Comment by emilvincentz** [comment]
Still seeing this on latest release along with other loading issues 

https://github.com/acterglobal/a3/assets/77483554/658e53b5-1346-4b95-ac9d-e477ffbba177

**Comment by emilvincentz** [comment]
It also ended up freezing the app, so had to restart which resolved it

**Comment by gnunicorn** [comment]
Okay, I have another piece of evidence:

it appears that our internal http-client doesn't limit the number of concurrent requests. And our UI issues a bunch of load stuff (e.g. user & room avatars, user counts, space sync ups) when we start up. And combined with the server rate limits that means a bunch of them are held open for a bit before they are responded to. On my local linux install with my larger account, I can see upwards of 250 pending concurrent requests with `lsof`:
```

acter   441431  ben 363u     IPv4            5162956       0t0      TCP slimbook:37972->h3002795.stratoserver.net:https (ESTABLISHED)
acter   441431  ben 364u     IPv4            5165974       0t0      TCP slimbook:38736->h3002795.stratoserver.net:https (ESTABLISHED)
```

It's fair to assume the number growth linear to the number of rooms the user is in. Thus if the numbers are even have way similar iOS is probably shutting us off with the IO error you are seeing as iOS has lower limits on open files per app.

**Comment by gnunicorn** [comment]
expected to be fixed with #1898
